src/ImageSharp/PixelFormats/IPixel.cs

```diff
@@ -14,14 +14,14 @@ namespace SixLabors.ImageSharp.PixelFormats;
 public interface IPixel<TSelf> : IPixel, IEquatable<TSelf>
     where TSelf : unmanaged, IPixel<TSelf>
 {
+#pragma warning disable CA1000 // Do not declare static members on generic types
```

Comment:
I'm going to open an issue for this on dotnet/roslyn-analyzers. Declaring static members on generic types is kind-of the purpose of `static abstracts in interfaces` 😆

Reply:
Looks like we have an issue already: https://github.com/dotnet/roslyn-analyzers/issues/6424

Reply:
I think the reason that exists is to avoid unnecessary binary size bloat for when you eg. accidentally declare members that don't actually depend on type parameters at all. We had a few of those in ImageSharp, where the bokeh blur processor had a couple static fields with some readonly buffers (like, constants), and a bunch of pixel agnostic processing methods. We saved a nice amount of binary size moving those out to a non-generic type. It would be nice if the analyzer was smarter and just avoided warning in cases where you do in fact use the type parameter(s), but kept warning otherwise.

Reply:
Nah, the analyzer exists not being of binary bloat but because `MyClass.Foo(x)` (given `Foo<T>(T x)`) works with generic inference while `MyClass<T>.Foo(x)` does not and since static members historically couldn't be `abstract` it didn't make sense to ever have them there.

So the analyzer really just hasn't been updated to account for the existence of `static abstract in interfaces`. Once it's been made aware, I'd expect it suggest such members be explicitly implemented on the type and to defer to the `MyClass.Foo<T>(...)` implementation.

Reply:
I’ve upvoted on the linked issue anyway